### ISSUE TITLE
Level should not show navbar when XP is not used in course

### ISSUE THREAD

**Issue by jrm-unamur** [FMCorz/moodle-block_xp#166]
Hi there,

When admin setting block_xp/navbardisplay is activated, the level is displayed in all courses even if the block is not present in course.
IMHO the level should be dipsplayed ONLY in courses where block_xp is activated

Kind regards

**Comment by FMCorz** [comment]
Hi @jrm-unamur,

Thanks for reporting this. You are right, the level is displayed in the navbar even for courses where the block was not added. This is indeed a bug. In the meantime, to mitigate this, you could use roles to remove/add the permission `block/xp:view` from the student role in the courses where XP should be hidden/shown.

**Comment by jrm-unamur** [comment]
Thanks for prompt answer Frédéric.
Do you plan to fix this in a next release? The permission trick is ok, but with many courses and few advanced teachers, it's not really easy to get them do this manipulation ;-)

Kind regards

Jean-Roch

**Comment by FMCorz** [comment]
Yes, we will fix it in the next release. I understand the workaround does not scale, I shared it just in case it could be applied. Thanks again!

**Comment by FMCorz** [comment]
Closing as fixed in v3.17.0
